### PR TITLE
Add admin page and new route option "onlyFor" to limit access to certain routes

### PR DESCRIPTION
**Pull request by robinmessage**

### CODE REVIEW

File: src/app/components/navigation/TrackedRoute.tsx [hunk 1]
```diff
@@ -23,12 +34,26 @@ const WrapperComponent = function({component: Component, trackingOptions, ...pro
...
-            return <WrapperComponent component={component} trackingOptions={trackingOptions} {...props} {...componentProps} />;
-        }} />;
+        if (rest.onlyFor !== undefined) {
+            const {onlyFor, user, ...rest$} = rest;
+            return <Route {...rest$} render={props => {
+                const propsWithUser = {user, ...props};
+                return <ShowLoading until={user}>
+                    {user && onlyFor(user) ?
+                        <WrapperComponent component={component} trackingOptions={trackingOptions} {...propsWithUser} />
+                        : <Redirect to="/login"/>
```
[the remark this answers is not in the document]
Comment (mlt47) [reply src/app/components/navigation/TrackedRoute.tsx]: I have pulled this out to a separate task on Trello, but is straight forward.

[the remark this answers is not in the document]
Comment (robinmessage) [reply src/app/components/navigation/TrackedRoute.tsx]: I figure it should probably look at loggedIn as well, and redirect to /login if there isn't even a loggedInUser, and forbidden if there is a logged in user?

File: src/app/components/navigation/TrackedRoute.tsx [hunk 2]
```diff
@@ -23,12 +34,26 @@ const WrapperComponent = function({component: Component, trackingOptions, ...pro
     return <Component {...props} />;
 };
 
-export const TrackedRoute = function({component, trackingOptions, componentProps, ...rest}: TrackedRouteProps) {
+const TrackedRouteComponent = function({component, trackingOptions, componentProps, ...rest}: TrackedRouteProps) {
```
[the remark this answers is not in the document]
Comment (robinmessage) [reply src/app/components/navigation/TrackedRoute.tsx]: Yeah, I couldn't see a neat way to split this up so I didn't. As you say, in the future!

File: src/IsaacAppTypes.tsx [hunk 3]
```diff
@@ -66,3 +78,5 @@ export interface ValidatedChoice<C extends ApiTypes.ChoiceDTO> {
 export function isValidatedChoice(choice: ApiTypes.ChoiceDTO|ValidatedChoice<ApiTypes.ChoiceDTO>): choice is ValidatedChoice<ApiTypes.ChoiceDTO> {
     return choice.hasOwnProperty("frontEndValidation");
 }
+
+export type LoggedInUser = {loggedIn: true} & ApiTypes.RegisteredUserDTO | {loggedIn: false};
```
[the remark this answers is not in the document]
Comment (robinmessage) [reply src/IsaacAppTypes.tsx]: Isn't it just!

**Pull request merged**